include/unifex/task.hpp

```diff
@@ -256,6 +257,12 @@ struct _task<T>::type {
     return *this;
   }
 
+  template <typename Fn, typename... Args>
+  friend type tag_invoke(
+      tag_t<co_invoke>, tag<type, Fn, Args...>, Fn fn, Args... args) {
```

Comment:
I think that you need to introduce extra `Fn2` and `Args2` template args here so that the subsequent args can be passed by value while allowing the original value category to differ from the argument type here (which we want to always be passed by-value).

I think you'll find that this formulation doesn't actually pass by-value if you pass an lvalue to `co_invoke()` as it will deduce `Fn` or `Args` to be an lvalue reference from the template parameters of `tag<...>` and thus deduce the argument type for `fn` and `args...` to be an lvalue.

Instead you need to use something like:
```
template<typename Fn, typename... Args, typename FnVal, typename... ArgsVal>
friend type tag_invoke(tag_t<co_invoke>, tag<type, Fn, Args...>, FnVal fn, ArgsVal... args) {
  co_return co_await std::invoke((Fn&&)fn, (Args&&)args...);
}
```

Although, as mentioned below I'm never been entirely comfortable with this preservation of value-category in this way after we've forcefully made a copy of the original argument. Why not just always pass an xvalue to the invocation?